flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/StateRequestBuffer.java

```diff
@@ -59,6 +60,9 @@ public StateRequestBuffer() {
     }
 
     void enqueueToActive(StateRequest<K, ?, ?> request) {
+        if (request.getRequestType() == StateRequestType.SYNC_POINT) {
+            request.getFuture().complete(null);
+        }
```

Comment:
If `requestType=StateRequestType.SYNC_POINT`, don't add the request to active buffer?

Reply:
Yes. Once the `SYNC_POINT` is about to add to the `activeQueue` (meaning that it is ready), complete it directly.